source/ShipModel.cpp

```diff
@@ -86,6 +86,24 @@ bool ShipModel::createAllTask(int data) {
 bool ShipModel::failAllTask() {
 	for (int i = 0; i < donuts.size(); i++) {
 		float angle = (float)(rand() % (int)(getSize()));
+		bool goodAngle = false;
+		while (!goodAngle) {
+			angle = (float)(rand() % (int)(getSize()));
+			for (unsigned int k = 0; k < breaches.size(); k++) {
+				float breachAngle = breaches[k]->getAngle();
+				float diff = shipSize / 2 - abs(abs(breachAngle - angle) - shipSize / 2);
+				if (breachAngle == -1 || diff > 15) {
+					goodAngle = true;
```

Comment:
Isn't the criteria to avoid all doors AND breaches? Here you set the condition to true if it avoids all doors OR avoids all breaches

Reply:
updated the code